app/controllers/api/v1/promotional_tweets_controller.rb

```diff
@@ -1,32 +1,32 @@
-class Api::V1::PromotionalTweetsController < ApplicationController
+class Api::V1::PromotionalTweetsController < Api::V1::BaseController
+  before_action :get_brand, only: [:create]
 
   def index
     @promotional_tweets = PromotionalTweet.all
   end
 
   def create
-    @create_promotional_tweet_service = CreatePromotionalTweet.new(image_params)
-    @promotional_tweet = @create_promotional_tweet_service.promotional_tweet
-    if @create_promotional_tweet_service.enqueue_process
-      render :show, status: :created
-    else
-      render json: { errors: @promotional_tweet.errors.messages }, status: :unprocessable_entity
+    binding.pry
+    image = params[:image]
+
+    ActiveRecord::Base.transaction do
+      @promotional_tweet = PromotionalTweet.create!(promo_tweet_params)
+      post_tweet_service = PostPromotionalTweet.new(promotional_tweet, image, @brand)
+      tweet_url = post_tweet_service.send!
+
+      if tweet_url.empty? 
+        raise ActiveRecord::Rollback
```

Comment:
You can raise any error here and it will still rollback. I'd recommend raising an `ApiErrors::StandardError` so that you can control the messaging to the client.